Plugins/Public/base_plugin/Main.cpp

```diff
@@ -45,6 +45,19 @@ set<uint> bannedSystemList;
 /// The ship used to construct and upgrade bases
 uint set_construction_shiparch = 0;
 
+/// Mininmum distances for base deployment
+bool enableDistanceCheck = false;
+float minMiningDistance = 30000;
+float minPlanetDistance = 2500;
+float minStationDistance = 10000;
+float minLaneDistance = 5000;
+float minJumpDistance = 15000;
+float minDistanceMisc = 2500;
+
+/// Deployment command cooldown trackimg
+unordered_map<uint, uint> deploymentCooldownMap;
+uint deploymentCooldownDuration = 60;
```

Comment:
we need a cooldown?

Reply:
Iteration on every solar and checking distance is not exactly a cheap operation, we don't want someone to be able to spam it and potentially lag out the server by inputting this several times a second via a macro.

Reply:
makes sense